olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -769,6 +901,74 @@ client::CancellationToken AuthenticationClient::Impl::SignOut(
   });
 }
 
+client::CancellationToken AuthenticationClient::Impl::IntrospectApp(
+    std::string access_token, IntrospectAppCallback callback) {
+  using ResponseType =
+      client::ApiResponse<IntrospectAppResult, client::ApiError>;
+
+  auto introspect_app_task =
+      [=](client::CancellationContext context) -> ResponseType {
+    if (!settings_.network_request_handler) {
+      client::ApiError result({static_cast<int>(http::ErrorCode::IO_ERROR),
+                               "Cannot introspect app while offline"});
+      return std::move(result);
+    }
+
+    client::AuthenticationSettings auth_settings;
+    auth_settings.provider = [&access_token]() { return access_token; };
```

Comment:
Do you need this? Probably not. If you leave this empty then the bearer token is not added to the headers.

Reply:
Yes, I need to authenticate  using access token somehow. I assume this is a valid way to do it for OlpClient::CallApi().

Reply:
At least the app/me is supposed to work without bearer. So if you remove it it is not working?

Reply:
app/me requires access token

Reply:
Yeah, you will get authetication error(401 I think).

Reply:
Ok. Then let's leave it like this.